axelor-business-production/src/main/java/com/axelor/apps/businessproduction/service/InvoicingProjectServiceBusinessProdImpl.java

```diff
@@ -101,4 +103,17 @@ public int countToInvoice(Project project) {
 
     return toInvoiceCount;
   }
+
+  @Transactional(rollbackOn = {AxelorException.class, Exception.class})
+  @Override
+  public InvoicingProject generateInvoicingProject(Project project) {
+
+    InvoicingProject invoicingProject = super.generateInvoicingProject(project);
+
+    if (invoicingProject.getId() == null && !invoicingProject.getManufOrderSet().isEmpty()) {
```

Comment:
Check not null for invoicingProject too , see super class method it can return null too.